--- origsrc/cairo-ocaml/src/Makefile	2006-02-09 09:57:11.000000000 -0600
+++ src/cairo-ocaml/src/Makefile	2008-01-09 01:00:14.890625000 -0600
@@ -40,7 +40,7 @@
 cairo.cmxa : $(call mloptobjs,$(cairo_SRC))
 	$(OCAMLMKLIB) -o cairo -oc mlcairo $^ $(CAIRO_LIBS)
 libmlcairo.a dllmlcairo.so : $(call cobjs,$(cairo_SRC))
-	$(OCAMLMKLIB) -o cairo -oc mlcairo $^ $(CAIRO_LIBS)
+	$(OCAMLMKLIB) -o cairo -oc mlcairo $^ -lbigarray $(CAIRO_LIBS)
 
 lablgtk_SRC = cairo_lablgtk.mli cairo_lablgtk.ml ml_cairo_lablgtk.c
 
@@ -49,7 +49,8 @@
 cairo_lablgtk.cmxa : $(call mloptobjs,$(lablgtk_SRC))
 	$(OCAMLMKLIB) -o cairo_lablgtk -oc mlcairo_lablgtk $^
 libmlcairo_lablgtk.a dllmlcairo_lablgtk.so : $(call cobjs,$(lablgtk_SRC))
-	$(OCAMLMKLIB) -o cairo_lablgtk -oc mlcairo_lablgtk $^
+	$(OCAMLMKLIB) -o cairo_lablgtk -oc mlcairo_lablgtk $^ \
+		-ldopt -L. -L$(C_LABLGTKDIR) -llablgtk2 -lmlcairo $(GDK_LIBS)
 
 svgcairo_SRC = svg_cairo.mli svg_cairo.ml ml_svg_cairo.c
 
@@ -58,7 +59,7 @@
 svg_cairo.cmxa : $(call mloptobjs,$(svgcairo_SRC))
 	$(OCAMLMKLIB) -o svg_cairo -oc mlsvgcairo $^ $(LIBSVG_CAIRO_LIBS)
 libmlsvgcairo.a dllmlsvgcairo.so : $(call cobjs,$(svgcairo_SRC))
-	$(OCAMLMKLIB) -o svg_cairo -oc mlsvgcairo $^ $(LIBSVG_CAIRO_LIBS)
+	$(OCAMLMKLIB) -o svg_cairo -oc mlsvgcairo $^ -ldopt -L. -lmlcairo $(LIBSVG_CAIRO_LIBS)
 
 $(call cobjs,$(cairo_SRC))    : CPPFLAGS+=$(CAIRO_CFLAGS)
 $(call cobjs,$(lablgtk_SRC))  : CPPFLAGS+=$(GDK_CFLAGS) -I$(C_LABLGTKDIR)
